src/models/giveaway.py

```diff
@@ -0,0 +1,50 @@
+from sqlalchemy import Column, Integer, Float, ForeignKey, String
+from sqlalchemy.orm import relationship
+from src.database import Base
+from src.models.user import User
+
+
+class Giveaway(Base):
+    """
+    A giveaway object.
+
+    Attributes:
+        - `id`              The ID of the giveaway.
+        - `name`            The name of the giveaway.
+        - `user_ids`        (nullable) The IDs of users entered into this giveaway.
+    """
+
+    __tablename__ = "giveaway"
+
+    id = Column(Integer, primary_key=True)
+    name = Column(String, nullable=False)
+    user_ids = relationship("GiveawayInstance", back_populates="users")
+
+    def __init__(self, **kwargs):
+        self.id = kwargs.get("id")
+        self.name = kwargs.get("name")
+        self.user_ids = kwargs.get("user_id")
+
+    def serialize(self):
+        return {"id": self.id, "name": self.name, "user_ids": self.user_ids}
```

Comment:
```suggestion
    users = relationship("User", secondary="giveaway_instance", back_populates="giveaways")
```

I just realized that `serialize` isn't used for anything so we could probably remove it (we should remove it for the other models as well). I'm pretty sure we can also remove the initializer too.

When using an association table, you want to use `secondary`. See the change above.